### ISSUE TITLE
Не работает чтение env('API_TOKEN'), вместо него возвращается NULL при наличии этого параметра в файле .env

### ISSUE THREAD

**Issue by jehkinen** [generalov-viktar/test-laravel#3]
https://github.com/generalov-viktar/test-laravel/blob/39224d59a3e21050a960ed86b06b402039bdaa15/app/Http/Middleware/ApiMiddleware.php#L18

Нельзя читать из энвов, потому что конфиги закэшированы через php artisan config, все параметры из энвов нужно брать через config
Добавить например новый ключ в config/app.php и назвать его 'api_key' =>env('API_KEY', null) например, какой смысл тогда вообще конфиги кэшировать?
Читать через config('app.api_key') где нужно уже в приложении

**Comment by jehkinen** [comment]
Лучше проверять сначала наиболее вероятное условие, а именно что апи кей валиден, проверка сначала негативных условий увеличивает количество проверок и срабатываний.

В данном случае можно просто сделать
f ($request->bearerToken() === config('app.api_key')) { 
 next()...
}

throw InvalidTokenException()....

**Comment by generalov-viktar** [comment]
Если посмотреть как записывается API_TOKEN в команде php artisan:generateToken, то можно увидеть, что после записи происходит кэширование нового конфига. 
Поэтому ключ считается новый при проверке.

Я исходил из того, что в тестовом задании было написано - "поместить ключ в .env".

**Comment by jehkinen** [comment]
Но все енвы через конфиги передаются, то что ты его . env поместил это абсолютно правильно, но далее энвы всегда передаются через конфиги, иначе они не будут закэшированы и более того, если заккшировать конфиги, то этот ключ не считается из энвов

**Comment by jehkinen** [comment]
https://github.com/alexeymezenin/laravel-best-practices/blob/master/russian.md#%D0%BD%D0%B5-%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%B0%D0%B9%D1%82%D0%B5-%D1%81-%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D0%BC%D0%B8-%D0%B8%D0%B7-%D1%84%D0%B0%D0%B9%D0%BB%D0%B0-env-%D0%BD%D0%B0%D0%BF%D1%80%D1%8F%D0%BC%D1%83%D1%8E

**Comment by generalov-viktar** [comment]
Да согласен. Мне исправить все текущие проблемы и залить в репозиторий исправления? Или это уже для самостоятельной работы над ошибками?

**Comment by jehkinen** [comment]
Можно ещё побороться =) прочитай плз laravel best practice и вот это https://github.com/adelf/acwa_book_ru
Так же стоит поправить все критичные ошибки и в части с конвертацией советую написать хоть пару юнитов, чтобы быть уверенным что нет ошибок и потерь денег при конвертации, это очень больно и критично, когда работаешь с реальными деньгами

**Comment by jehkinen** [comment]
Да, можно ещё побороться =) и прочитай плз laravel best practice и книгу и можно будет провести собес вначале следующей неделе, мне очень понравилось что код аккуратный

**Comment by generalov-viktar** [comment]
Хорошо, договорились. Но я буду в отпуске с 15 числа по 26. Устроит тебя если после отпуска я сделаю изменения? 
Просто лечу с женой на Байкал, а там возможности не будет исправить все. Отпуск давно хотел))

**Comment by jehkinen** [comment]
Думаю да, если не найдём кого-то ещё за это время, то велком)

**Comment by generalov-viktar** [comment]
Договорились)

**Comment by generalov-viktar** [comment]
Я еще оберну все тестами, в докер засуну и инструкцию напину подробную.

**Comment by generalov-viktar** [comment]
Привет, актуальна еще вакансия? Я вернулся из отпуска.
